### ISSUE TITLE
Possible regression in error message for enum comparison

### ISSUE THREAD

**Issue by patriksvensson** [rust-lang/rust#77279]
It was a little while since I wrote any Rust and noticed something when I wrote some code today.
I'm _not sure_ if I remember incorrectly but didn't comparison with an `enum` give a rather helpful message that the enum don't implement `PartialEq` or something like that?

Perhaps this is intentional? In that case, feel free to close this issue.

**I tried this code:**

```rust
pub enum Corgi {
  Foo,
  Bar
}

fn my_func(lol: Corgi ) {
  if lol == Corgi::Foo {
    println!("Foo!");
  }
}
```

**I expected to see this happen:**
I expected to get an error about `Corgi` not implementing `PartialEq`.

**Instead, this happened:**
I get the error `error[E0369]: binary operation '==' cannot be applied to type `

### Meta

`rustc --version --verbose`:
```
rustc 1.46.0 (04488afe3 2020-08-24)
binary: rustc
commit-hash: 04488afe34512aa4c33566eb16d8c912a3ae04f9
commit-date: 2020-08-24
host: x86_64-pc-windows-msvc
release: 1.46.0
LLVM version: 10.0
```

**Comment by SNCPlay42** [comment]
The full error output is:

```
   Compiling playground v0.0.1 (/playground)
error[E0369]: binary operation `==` cannot be applied to type `Corgi`
 --> src/lib.rs:7:10
  |
7 |   if lol == Corgi::Foo {
  |      --- ^^ ---------- Corgi
  |      |
  |      Corgi
  |
  = note: an implementation of `std::cmp::PartialEq` might be missing for `Corgi`

error: aborting due to previous error

For more information about this error, try `rustc --explain E0369`.
error: could not compile `playground`.

To learn more, run the command again with --verbose.
```

The note does bring up `PartialEq`.

**Comment by patriksvensson** [comment]
@SNCPlay42 Hmmm, I'm not seeing a `note:` section at all.

![image](https://user-images.githubusercontent.com/357872/94395722-97312280-0160-11eb-9aae-a0862f259ece.png)

**Comment by lzutao** [comment]
Try nightly or beta toolchain.

**Comment by patriksvensson** [comment]
@lzutao I tried the nightly toolchain like issue template mentioned and getting the same result there. Just tried the beta toolchain and same thing there.

**Comment by patriksvensson** [comment]
The reproduction I provided doesn't work. I have no idea why I'm not seeing the `note:` section above, but closing for now until I can find a proper reproduction of the problem.

![image](https://user-images.githubusercontent.com/357872/94396809-01e35d80-0163-11eb-985a-e8903a81bbfb.png)

**Comment by bugadani** [comment]
I tried to reproduce the issue on the exact same host/version, using Windows Terminal, Powershell, cmd's command line and VSCode's terminal, but all of these included the expected note in the output. I've even tried multiple oh-my-posh themes just to see if those interfere with the output, but no luck. Would you mind sharing your _entire_ terminal setup?

![image](https://user-images.githubusercontent.com/977627/94437827-9323f580-019e-11eb-8af0-478356e8b231.png)
